### ISSUE TITLE
How to properly get JSONSchema for Fluent Model

### ISSUE THREAD

**Issue by joshdholtz** [mattpolzin/VaporOpenAPI#8]
Hello 👋 

First of all, thank you for making this! I'm creating a new API and I wanted to make sure I had doc from the start. This is so great 😊 

I'm having troubles getting my Fluent model to properly convert into a `JSONSchema` for the docs. I've attached my code and an example of the docs below. In the example screenshot, you can see that my sample response had underscores in front of it and some other values that I believe are from serializing the Fluent model. 

Do you happen to know how I can fix this to still use my Fluent model to make a `JSONSchema` that wouldn't include any of private properties (assuming that is what the underscores are) but instead looks like the normal JSON response that a model should make? 😇 

Sorry if not all of this makes sense. I'm still pretty new to Vapor 🙃 

---

```swift
extension AuthController {
    struct AuthEmailContext: RouteContext {
        typealias RequestBodyType = EmailVerificationAttempt.Create

        static let defaultContentType: HTTPMediaType? = nil
        static let shared = Self()

        let success: ResponseContext<EmailVerificationAttempt> = .init { response in
            response.headers.contentType = .json
            response.status = .created
        }
    }
}
```

```swift
extension EmailVerificationAttempt: OpenAPIEncodedSchemaType {
    static func openAPISchema(using encoder: JSONEncoder) throws -> JSONSchema {
        return try genericOpenAPISchemaGuess(using: encoder)
    }
}
```

<img width="1381" alt="Screen Shot 2021-06-29 at 8 03 52 PM" src="https://user-images.githubusercontent.com/401294/123886460-2304f480-d915-11eb-8e67-b24656286851.png">

**Comment by mattpolzin** [comment]
What you are seeing, as you've surmised, is that the `genericOpenAPISchemaGuess()` function simply crawls all of the properties exposed by a type. In the case of a Fluent model, these properties are not directly representative of how the Fluent model is encoded/decoded because Fluent models have custom `Codable` conformances.

At this point, I don't have code written that takes a Fluent model and produces appropriate API documentation. That means that aside from writing an alternative schema generator that specifically uses Fluent models, there are a couple of options:
1. Write the request/response schemas by hand, just taking advantage of the `VaporOpenAPI` library's ability to fill in details about routes, content types, and parameters for you.
2. Use simple `structs` to encode/decode your API response bodies and request bodies and then in turn write the glue code that takes a decoded struct and modifies a Fluent model or takes a Fluent model and creates a new struct that can be encoded for an API response body.

The second option, though more work, is the one I have often opted for. Unless you are willing to let the default encode/decode of Fluent models dictate your API's interface, it's the only way to go anyway.

Also, If someone wanted to write a schema generator that took Fluent models and used information about how they are structured to produce valid documentation on their encoding and decoding, I would definitely accept that as a nice feature of this library!

The bottom line (unfortunately for what you are hoping to accomplish) is that `genericOpenAPISchemaGuess()` is only smart enough to look at the fields it has in front of it and because Fluent models use features like Swift's PropertyWrappers the fields it sees are not the same structure nor do they have the same names as the encoded form of the same Fluent model.

**Comment by joshdholtz** [comment]
@mattpolzin Thank you so much for the response! This was _kind of_ my guess but I just wanted to make sure since I’m pretty new to all of this 🙃 I think making structs to encode/decode the API response makes a lot more sense. I really don’t want (and shouldn’t) tie my model directly to request/response. I’ll go with that for now but I might also try to make that fluent model PR if I get bored one day 😉

**Comment by JaapWijnen** [comment]
Would you have an example struct of a model with a fluent relation? I'm not sure how to translate between those two worlds
